assets/src/edit-story/components/canvas/utils/useSnapping.js

```diff
@@ -14,34 +14,100 @@
  * limitations under the License.
  */
 
+/**
+ * External dependencies
+ */
+import { useCallback } from 'react';
+
 /**
  * Internal dependencies
  */
 import { useGlobalIsKeyPressed } from '../../keyboard';
 import useCanvas from '../useCanvas';
 
 function useSnapping({ canSnap, otherNodes }) {
-  const { canvasWidth, canvasHeight } = useCanvas(
+  const {
+    canvasWidth,
+    canvasHeight,
+    pageContainer,
+    canvasContainer,
+    designSpaceGuideline,
+    showSafeZone,
+  } = useCanvas(
     ({
       state: {
         pageSize: { width: canvasWidth, height: canvasHeight },
+        pageContainer,
+        canvasContainer,
+        designSpaceGuideline,
+        showSafeZone,
       },
-    }) => ({ canvasWidth, canvasHeight })
+    }) => ({
+      canvasWidth,
+      canvasHeight,
+      pageContainer,
+      canvasContainer,
+      designSpaceGuideline,
+      showSafeZone,
+    })
   );
 
   // ⌘ key disables snapping
   const snapDisabled = useGlobalIsKeyPressed('meta');
   canSnap = canSnap && !snapDisabled;
+
+  const toggleDesignSpace = useCallback(
+    (visible) => {
+      designSpaceGuideline.style.visibility = visible ? 'visible' : 'hidden';
+    },
+    [designSpaceGuideline]
+  );
+  const handleSnap = useCallback(
+    ({ elements }) =>
+      // Show design space if we're snapping to any of its edges
+      toggleDesignSpace(
```

Comment:
When testing, I saw the design space snaplines being shown while drop targeting as well. Assuming this is part of one of the todo-s? Or perhaps that was happening already before?
![droptargeting](https://user-images.githubusercontent.com/3294597/94432479-1e00f200-0197-11eb-84db-36d86c824c39.gif)

Reply:
Hm, all snapping guidelines should really be disabled - you also see the guideline for page center there, which is also misleading. Can we do that dynamically during drag?

Reply:
Maybe our good friend @daybrush can help us out here.

Reply:
Perhaps if there's an `activeDropTargetId`, snapping could be disabled?

Reply:
Hey, that worked surprisingly well, thanks! Done in 17e9d47a6.